
The protein folding problem remains one of the outstanding challenges
in structural biology \cite{dill2012folding}.  It is usually defined
as the task of predicting the three-dimensional (3D) structure of a
protein from its amino acid sequence. A crucial step when making such
prediction is the ranking of candidate structures. Given a number of
structural models, can we computationally predict how close each of
them is to the unknown native fold of the protein?

Progress in the field is monitored through the Critical Assessment of
protein Structure Prediction (CASP) competition \cite{moult1995large},
in which protein folding methods are evaluated in terms of their
accuracy at predicting protein structures ahead of their
publication. Most methods in competition include a conformational
sampling step, which generates a number of plausible protein
conformations, and a quality assessment step, which attempts to select
the conformations closest to the unknown native structure.

In this work we explore the application of deep learning to the
problem of ``model quality assessment'' (MQA), also called
``estimation of model accuracy'' (EMA) \cite{kryshtafovych2015}. Deep
learning (DL) has recently garnered considerable interest in the
research community \cite{lecun2015deep}, particularly in computer
vision and natural language processing. Unlike more ``shallow''
machine learning approaches, deep learning improves performance by
learning a hierarchical representation of the raw data at hand. It
alleviates the need for feature engineering, which has traditionally
constituted the bulk of the work done by researchers.

Deep learning has recently been applied to biological data and has
yielded remarkable results for predicting the effects of genetic
variations on human RNA splicing \cite{xiong2015human}, for
identifying DNA- and RNA-binding
motifs \cite{alipanahi2015predicting}, and for predicting the effects
of non-coding DNA variants with single nucleotide
precision \cite{zhou2015predicting}. These successes have one thing in
common: they use raw data directly as input and do not attempt to
engineer features from them.

DL-inspired methods have been used for protein structure quality
assessment \cite{nguyen2014dlpro, cao2016deepqa,
uziela2017proq3d}. For instance, DeepQA \cite{cao2016deepqa} uses 9
scores from other MQA methods and 7 physico-chemical features
extracted from the structure as input features to a deep restricted
Boltzmann machine \cite{hinton2006fast}. The method has been
reported \cite{cao2016deepqa} to outperform ProQ2 \cite{ray2012proq2},
which was the top-performing method in the CASP11
competition \cite{kryshtafovych2015}.  ProQ3D \cite{uziela2017proq3d}
uses the same high-level input features as the earlier ProQ3
method \cite{uziela2016proq3} (including the
Rosetta \cite{leaverfay2011rosetta} score) but achieves better
performance by replacing the support vector machine model by a deep
neural network. Since the original ProQ3 method had one of the top
performances in CASP12 \cite{elofsson2017qacasp12}, it can be expected
that ProQ3D performs equally well. Although both DeepQA and ProQ3D
methods are based on deep neural networks, they use high-level
features as input. In that sense, they use DL models more as
traditional ``shallow'' classifiers than as end-to-end learning
models. It is likely that they do not get all the advantages offered
by the DL approach.
%
By comparison, the DL-Pro algorithm \cite{nguyen2014dlpro} uses a
sligthly more raw input, consisting of the eigenvectors of the
C$\alpha$-to-C$\alpha$ distance matrix. The model itself is an
autoencoder \cite{hinton2006reducing} trained to classify the
structures into either ``near native'' or ``not near native''.

More in line with the ``end-to-end'' spirit of deep learning, methods
using as input a 3D representation of the structure have been
developed to score protein-ligand poses \cite{wallach2015atomnet,
ragoza2017protein}, to predict ligand-binding protein
pockets \cite{jimenez2017deepsite}, and to predict the effect of a
protein mutation \cite{torng2017}. The molecules of interest are
treated as 3D objects represented on a grid and the predictions are
obtained from that information only. While a rigorous comparison of
each of these methods is not always possible, they generally appear to
improve on the state of the art: Both
AtomNet \cite{wallach2015atomnet} and the 3D convolutional neural
network of Ragoza et al.\ \cite{ragoza2017protein} perform
consistently better than either Smina \cite{koes2013smina} or AutoDock
Vina \cite{trott2009vina}.
%
For small molecules, Sch\"{u}tt and coworkers \cite{schutt2017quantum,
schutt2017moleculenet} have developed deep neural networks to predict
the molecular energy of a variety of chemical compounds in various
conformations (or even various isomeric states). These models,
intended to be used as universal force fields, are trained on ab
initio energies and forces, and use only the nuclear charges and the
interatomic distance matrix as input.
